shop/src/main/java/ro/msg/learning/shop/Service/Implementation/ProductService.java

```diff
@@ -0,0 +1,111 @@
+package ro.msg.learning.shop.Service.Implementation;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.DTO.ProductAndProductCategoryDto;
+import ro.msg.learning.shop.Domain.Product;
+import ro.msg.learning.shop.Domain.ProductCategory;
+import ro.msg.learning.shop.Exception.ProductAlreadyExistingException;
+import ro.msg.learning.shop.Exception.ProductCategoryNotFoundException;
+import ro.msg.learning.shop.Exception.ProductNotFoundException;
+import ro.msg.learning.shop.Mapper.ProductAndProductCategoryTranslatorMapper;
+import ro.msg.learning.shop.Repository.IProductRepository;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+public class ProductService {
+    @Autowired
+    private ProductCategoryService productCategoryService;
+    @Autowired
+    private IProductRepository productRepository;
+    @Autowired
+    private LocationService locationService;
+    @Autowired
+    private StockService stockService;
+
+
+    public ProductAndProductCategoryDto createProduct(ProductAndProductCategoryDto productAndProductCategoryDto) throws Exception {
+        //Validate the miss of the product in the product list
+        if (productRepository.existsByAttributes(
+                productAndProductCategoryDto.getProductName(),
+                productAndProductCategoryDto.getProductDescription(),
+                productAndProductCategoryDto.getProductPrice(),
+                productAndProductCategoryDto.getProductWeight(),
+                productAndProductCategoryDto.getProductSupplier(),
+                productAndProductCategoryDto.getProductImageUrl(),
+                productAndProductCategoryDto.getProductCategoryId())) {
+            throw new ProductAlreadyExistingException(productAndProductCategoryDto.getProductName());
+        }
+        //Validate the existence of the product category in the product list
+        ProductCategory productCategory = this.productCategoryService.findProductCategoryById(productAndProductCategoryDto.getProductCategoryId());
```

Comment:
Here we will need the already implemented method `findById()` which returns `Optional<ProductCategory>`